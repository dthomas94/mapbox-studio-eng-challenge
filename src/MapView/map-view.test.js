import React from "react";
import { MapView } from ".";
import { mount } from "enzyme";
import { mountToJson } from "enzyme-to-json";

const mockInstantiationTracker = jest.fn();

jest.mock("mapbox-gl", () => {
	return {
		Map: class {
			constructor({ container }) {
				mockInstantiationTracker();
			}

			on(eventName, layer, callbackFn) {
				mockInstantiationTracker();
			}
		},
	};
});

describe("MapView", () => {
	let wrapper;
	beforeEach(() => {
		jest.clearAllMocks();
		wrapper = mount(<MapView favorites={[]} onMarkerClick={jest.fn()} />);
	});
	it("renders", () => {
		expect(mountToJson(wrapper)).toMatchSnapshot();
	});
	it("initializes mapbox gl", () => {
		expect(mockInstantiationTracker).toHaveBeenCalledTimes(2);
	});
});
