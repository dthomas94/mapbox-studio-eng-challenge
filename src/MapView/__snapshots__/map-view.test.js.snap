// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapView renders 1`] = `
<MapView
  favorites={Array []}
  onMarkerClick={[MockFunction]}
>
  <div
    className="map-container"
  />
</MapView>
`;
